crates/red_knot/src/cli/verbosity.rs

```diff
@@ -0,0 +1,31 @@
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) enum VerbosityLevel {
+    Info,
+    Debug,
+    Trace,
+}
+
+/// Logging flags to `#[command(flatten)]` into your CLI
+#[derive(clap::Args, Debug, Clone, Default)]
+#[command(about = None, long_about = None)]
+pub(crate) struct Verbosity {
+    #[arg(
+        long,
+        short = 'v',
+        help = "Use verbose output",
+        action = clap::ArgAction::Count,
+        global = true,
+    )]
+    verbose: u8,
+}
+
+impl Verbosity {
+    pub(crate) fn level(&self) -> Option<VerbosityLevel> {
+        match self.verbose {
+            0 => None,
+            1 => Some(VerbosityLevel::Info),
+            2 => Some(VerbosityLevel::Debug),
+            _ => Some(VerbosityLevel::Trace),
+        }
+    }
```

Comment:
I feel like this could be simplified if you added a `VerbosityLevel::Warn` variant and just returned `VerbosityLevel` from this method, rather than `Option<VerbosityLevel>`. Something like this?

```diff
diff --git a/crates/red_knot/src/cli/verbosity.rs b/crates/red_knot/src/cli/verbosity.rs
index 30bc2a4a5..470bfa49c 100644
--- a/crates/red_knot/src/cli/verbosity.rs
+++ b/crates/red_knot/src/cli/verbosity.rs
@@ -1,5 +1,6 @@
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
 pub(crate) enum VerbosityLevel {
+    Warn,
     Info,
     Debug,
     Trace,
@@ -19,12 +20,12 @@ pub(crate) struct Verbosity {
 }
 
 impl Verbosity {
-    pub(crate) fn level(&self) -> Option<VerbosityLevel> {
+    pub(crate) fn level(&self) -> VerbosityLevel {
         match self.verbose {
-            0 => None,
-            1 => Some(VerbosityLevel::Info),
-            2 => Some(VerbosityLevel::Debug),
-            _ => Some(VerbosityLevel::Trace),
+            0 => VerbosityLevel::Warn,
+            1 => VerbosityLevel::Info,
+            2 => VerbosityLevel::Debug,
+            _ => VerbosityLevel::Trace,
         }
     }
 }
diff --git a/crates/red_knot/src/main.rs b/crates/red_knot/src/main.rs
index 35bfe2380..f39a23fd2 100644
--- a/crates/red_knot/src/main.rs
+++ b/crates/red_knot/src/main.rs
@@ -75,7 +75,7 @@ pub fn main() -> anyhow::Result<()> {
     } = Args::parse_from(std::env::args().collect::<Vec<_>>());
 
     let verbosity = verbosity.level();
-    countme::enable(verbosity == Some(VerbosityLevel::Trace));
+    countme::enable(verbosity >= VerbosityLevel::Trace);
     setup_tracing(verbosity);
 
     let cwd = if let Some(cwd) = current_directory {
@@ -134,13 +134,13 @@ pub fn main() -> anyhow::Result<()> {
 }
 
 struct MainLoop {
-    verbosity: Option<VerbosityLevel>,
+    verbosity: VerbosityLevel,
     orchestrator: crossbeam_channel::Sender<OrchestratorMessage>,
     receiver: crossbeam_channel::Receiver<MainLoopMessage>,
 }
 
 impl MainLoop {
-    fn new(verbosity: Option<VerbosityLevel>) -> (Self, MainLoopCancellationToken) {
+    fn new(verbosity: VerbosityLevel) -> (Self, MainLoopCancellationToken) {
         let (orchestrator_sender, orchestrator_receiver) = crossbeam_channel::bounded(1);
         let (main_loop_sender, main_loop_receiver) = crossbeam_channel::bounded(1);
 
@@ -203,12 +203,12 @@ impl MainLoop {
                 }
                 MainLoopMessage::CheckCompleted(diagnostics) => {
                     eprintln!("{}", diagnostics.join("\n"));
-                    if self.verbosity == Some(VerbosityLevel::Trace) {
+                    if self.verbosity >= VerbosityLevel::Trace {
                         eprintln!("{}", countme::get_all());
                     }
                 }
                 MainLoopMessage::Exit => {
-                    if self.verbosity == Some(VerbosityLevel::Trace) {
+                    if self.verbosity >= VerbosityLevel::Trace {
                         eprintln!("{}", countme::get_all());
                     }
                     return;
@@ -361,12 +361,12 @@ enum OrchestratorMessage {
     FileChanges(Vec<FileWatcherChange>),
 }
 
-fn setup_tracing(verbosity: Option<VerbosityLevel>) {
+fn setup_tracing(verbosity: VerbosityLevel) {
     let trace_level = match verbosity {
-        None => Level::WARN,
-        Some(VerbosityLevel::Info) => Level::INFO,
-        Some(VerbosityLevel::Debug) => Level::DEBUG,
-        Some(VerbosityLevel::Trace) => Level::TRACE,
+        VerbosityLevel::Warn => Level::WARN,
+        VerbosityLevel::Info => Level::INFO,
+        VerbosityLevel::Debug => Level::DEBUG,
+        VerbosityLevel::Trace => Level::TRACE,
     };
```

Reply:
Yeah, I started with this but I disliked the name warn for the default. What the option indicates is that the user didn't specify any verbosity. I considered using `DEFAULT`, but then went with `Option`. 

We probably need to revisit this once we add other means of logging messages to the user.

Reply:
Maybe `Unspecified`? Right now it feels unclear to me, as a reader of the code, why it returns `Option`, so I'd at least add a comment explaining the decision, I think :-)